### PR TITLE
fix/feat-forgot-password-added

### LINKED ISSUE #207: Feat: Forgot password feature.
### Describe the feature

In sign in , adding forgot password feature or create a new password feature will enhance the user experience .

### Add ScreenShots

Screenshot (478)

### Record

- [X] I agree to follow this project's Code of Conduct
- [X] I'm a GSSOC'24 contributor
- [X] I want to work on this issue

### PR DESCRIPTION
**Pull request by Adityajparmar37**
Forgot Password Feature added

Fixes #207 

## PLEASE NOTED
IN ENV VARIAVLE FILE (.env) added following things without it feature will not work neither can be check

**1) **EMAIL**="Added your email address";
2) **MAILPASS**="Add password for allowing mailing outside gmail mail service";**

## To GENERATE MAILPASS
https://myaccount.google.com/apppasswords?rapt=AEjHL4PAhcbtFEpLwfNtVix3bfiGe71GdrW_Naiuvp_NVnMZyTd0UR07M2mVnEyWzkw9kB99YVhhfEVtjxTi3QWSZ39biK-zGwnghm0u778vwmlh6TFbmh4

**Go to this link and create *New APP PASSWORD*** 

## NOTE
***In https://myaccount.google.com/security, do you see 2-step verification set to *ON****

## Type of change
- [X] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?
Yes for both student and canteen 

# Screenshorts and Vedios:
https://github.com/VanshKing30/FoodiesWeb/assets/105008414/5ffb632e-741f-4b56-b3b3-17966975c5ef

# Checklist:
give a X on it which is applicable

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] My changes generate no new warnings
- [X] New and existing unit tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by Adityajparmar37** [comment]
Please review the code

**Review by aayushchugh — CHANGES_REQUESTED** [review]
Can you please update the required `.env` variables in `README` too?

**Comment by Adityajparmar37** [comment]
> Can you please update the required `.env` variables in `README` too?

I can but the thing is i cannot give my EMAIL or MAILPASS key

**Comment by Adityajparmar37** [comment]
I have added .env in README file please refer it

**Comment by hustlerZzZ** [comment]
> I have added .env in README file please refer it

Kindly resolve the conflicts... @Adityajparmar37

**Review by hustlerZzZ — CHANGES_REQUESTED** [review]
Resolve the conflicts

File: README.md [hunk 1]
```diff
@@ -62,30 +86,37 @@ Welcome to Foodies, your go-to college dining companion! Designed for seamless c
...
-5. **Start the Backend Server (in the terminal within the /server directory):** <br>
+
+   ### **ENV Variables**
+   Setup MongoDB local host instance and port in env file in your local device. Default port is 4000
+   Sample uri is given below.
+   ```
+   PORT=3000
+   DATABASE_URL="http://localhost:21713/foods"
+   EMAIL="your or website email address to send link"
+   MAILPASS="mail pass key ( app password)"
```
Comment (aayushchugh) [inline README.md:112]: Change this too

```
EMAIL="The email from which forgot the password email will be sent"
MAILPASS="password for the email ( app password )"
```

and provide a link to generate an app password too

**Review by aayushchugh — CHANGES_REQUESTED** [review]
Reset looks good, amazing job @Adityajparmar37 💪🏼

**Review by aayushchugh — APPROVED** [review]
LGTM 🚀

**Comment by Adityajparmar37** [comment]
Branch conflicts has been resolve plz review the code again

**Review by hustlerZzZ — APPROVED** [review]
Great Work Buddy

**Comment by Adityajparmar37** [comment]
Hey , can you please check why production or deployment has been failed . 

image


Please check it

**Pull request merged**